scripts/simplify.py

```diff
@@ -0,0 +1,215 @@
+	for attribute in keys:
+		element.removeAttribute(attribute)
+
+	# Sort
+	attributes.sort(key=lambda name: name[0])
+
+	# Add attributes back
+	for name, value in attributes:
+		element.setAttribute(name, value)
+
+def process_image_layer(xml, layer, new_layer):
+	images = layer.getElementsByTagName('image')
+
+	image_allowed_attrs = ['xlink:href', 'y', 'x', 'preserveAspectRatio', 'height', 'width']
+	for image in images:
+		new_image = xml.createElement('image')
+
+		# Recreate each allowed attribute in the newly created element
+		for allowed_attr in image_allowed_attrs:
+			# Check if field has a value
+			if not image.getAttribute(allowed_attr):
+				continue
+		
+			new_image.setAttribute(allowed_attr, image.getAttribute(allowed_attr))
+
+		# Add UUID if needed
+		if not new_image.getAttribute('pcbre:uuid'):
+			new_image.setAttribute('pcbre:uuid', str(uuid.uuid4()))
+
+		# Sort attributes
+		sort_element_attr(new_image)
+
+		new_layer.appendChild(new_image)
+
+	# Sort elements
+	new_layer.childNodes.sort(key=lambda x: x.getAttribute('pcbre:uuid'))
+
+def process_zones_layer(xml, layer, new_layer):
+	zones = layer.getElementsByTagName('path')
+
+	trace_allowed_attrs = ['d', 'transform', 'inkscape:connector-curvature', 'inkscape:label', 'pcbre:uuid']
+	for zone in zones:
+		new_zone = xml.createElement('path')
+
+		# Recreate each allowed attribute in the newly created element
+		for allowed_attr in trace_allowed_attrs:
+			# Check if field has a value
+			if not zone.getAttribute(allowed_attr):
+				continue
+		
+			new_zone.setAttribute(allowed_attr, zone.getAttribute(allowed_attr))
+
+		# TODO: Remove transforms
+
+		# Assign default zone style
+		new_zone.setAttribute('style', default_zone_style)
+
+		# Assign default zone label
+		if not new_zone.getAttribute('inkscape:label'):
+			new_zone.setAttribute('inkscape:label', default_zone_label)
+
+		# Add UUID if needed
+		if not new_zone.getAttribute('pcbre:uuid'):
+			new_zone.setAttribute('pcbre:uuid', str(uuid.uuid4()))
+
+		# Sort attributes
+		sort_element_attr(new_zone)
+
+		new_layer.appendChild(new_zone)
+
+	# Sort elements
+	new_layer.childNodes.sort(key=lambda x: x.getAttribute('pcbre:uuid'))
+
+def process_traces_layer(xml, layer, new_layer):
+	traces = layer.getElementsByTagName('path')
+
+	trace_allowed_attrs = ['d', 'inkscape:connector-curvature', 'pcbre:uuid']
+	for trace in traces:
+		new_trace = xml.createElement('path')
+
+		# Recreate each allowed attribute in the newly created element
+		for allowed_attr in trace_allowed_attrs:
+			# Check if field has a value
+			if not trace.getAttribute(allowed_attr):
+				continue
+		
+			new_trace.setAttribute(allowed_attr, trace.getAttribute(allowed_attr))
+
+		# Assign default trace style
+		new_trace.setAttribute('style', default_trace_style)
+
+		# Add UUID if needed
+		if not new_trace.getAttribute('pcbre:uuid'):
+			new_trace.setAttribute('pcbre:uuid', str(uuid.uuid4()))
```

Comment:
Instead of `uuid`, I'd just pick the first N comma-separated symbols of the `g` attribute, and hash them or spit out their numbers in some form. Usually a path starts with "m <x> <y>", so this would be the start location of the path.

If the path has been re-imported into the users SVG editor, this is irrelevant, because it will already have it's uuid (wether it's a hash, or some location we once chose).

But if the users working copy does not have the hashes (because they never re-imported after our tool generated hashes), the generated hashes should still match, unless the trace was moved (if you don't hash it, but just spit out the numbers, then it would even be stable if it was moved, as long as it wasn't moved too much).

Reply:
We probably don't even have to embed the hash/uuid/location in the file explicitly. We'd just have to specify this as sorting criteria (and use the existing `g` attribute).

Reply:
Ignore this; UUID is probably fine and less error prone.

I'll work on visual conflict resolution in compare.py which will tell us which path of `svg_new` overlaps with `svg_old`. If there's too much overlap we can throw away the new trace, even if it has a different name or a different path.